enlace/reports/num_programs.php

```diff
@@ -296,20 +272,40 @@ function(idx, display) {
     /* see results with month and year selected: */
 } else {
     ?>
-    <h3>Results as of <?php echo $_POST['month_select'] ?>/<?php echo $_POST['year_select'] ?></h3>
+    <?php
+        include "../include/dbconnopen.php";
+        $start_date_sqlsafe = mysqli_real_escape_string($cnnEnlace, $_POST['programs_start_date']);
+        $end_date_sqlsafe = mysqli_real_escape_string($cnnEnlace, $_POST['programs_end_date']);
+        $date_select_query_str = '';
+        if($start_date_sqlsafe != '') {
+            if ($end_date_sqlsafe != '') {
+                $results_string = "Results between $start_date_sqlsafe and $end_date_sqlsafe";
+                $date_select_query_str = "Session_Names.Start_Date BETWEEN '$start_date_sqlsafe' AND '$end_date_sqlsafe'
+                                          AND Session_Names.End_Date BETWEEN '$start_date_sqlsafe' AND '$end_date_sqlsafe'";
+            }
+            else {
+                $results_string = "Results after $start_date_sqlsafe";
+                $date_select_query_str = "Session_Names.Start_Date >= '$start_date_sqlsafe' AND
+                                          Session_Names.End_Date >= '$start_date_sqlsafe'";
+            }
+        }
+        else if ($end_date_sqlsafe != '') {
+            $results_string = "Results before $end_date_sqlsafe";
+            $date_select_query_str = "Session_Names.Start_Date <= '$end_date_sqlsafe'
+                                      Session_Names.End_Date <= '$end_date_sqlsafe'";
+        }
+    ?>
+    <h3><?php echo $results_string ?></h3>
     <table class="all_projects">
         <tr>
             <th>Program Name</th>
             <th>Session Name</th>
+            <th>Session End Date</th>

```

Comment:
If you're adding end date here (which I think is nice) then I think you should also include start date.

Reply:
Eddie said that they only care about the end date, and didn't want to clutter up the report.

Reply:
That's so funny.  Well, okay!